src/libpriv/rpmostree-core.c

```diff
@@ -3297,9 +3297,12 @@ apply_rpmfi_overrides (RpmOstreeContext *self,
       rpm_mode_t mode = rpmfiFMode (fi);
       rpmfileAttrs fattrs = rpmfiFFlags (fi);
       const gboolean is_ghost = fattrs & RPMFILE_GHOST;
+      const gboolean has_non_bare_user_mode =
```

Comment:
Ahhh OK I see the issue now. Hmm, this is pretty subtle. How about a comment like:

```c
/* if we hardlinked from a bare-user repo, we won't have those higher bits set otherwise */
```

?